compiler/src/dotty/tools/dotc/semanticdb/Tools.scala

```diff
@@ -12,8 +12,10 @@ object Tools:
   def mkURIstring(path: Path): String =
     // Calling `.toUri` on a relative path will convert it to absolute. Iteration through its parts instead preserves
     // the resulting URI as relative.
-    val uriParts = for part <- path.asScala yield new java.net.URI(null, null, part.toString, null)
-    uriParts.mkString("/")
+    // To prevent colon `:` from being treated as a scheme separator, prepend a slash `/` to each part to trick the URI
+    // parser into treating it as an absolute path, and then strip the spurious leading slash from the final result.
+    val uriParts = for part <- path.asScala yield new java.net.URI(null, null, "/" + part.toString, null)
```

Comment:
looks like this URI constructor is just not the right tool  for the job...

Reply:
It's a poor fit for sure, but sadly the Java standard library doesn't provide anything better. I considered copy-pasting the OpenJDK private method on `java.net.URI` that does the quoting, but it's, uh, [_very_ Java](https://github.com/openjdk/jdk/blob/802ef38060080254e55621e4c64fa31a6c0b7b18/src/java.base/share/classes/java/net/URI.java#L2772).

Reply:
its the only way to parse URI as the URI parser is private to the class